pallets/automation-time/src/tests.rs

```diff
@@ -663,23 +898,94 @@ fn trigger_tasks_completes_some_native_transfer_tasks() {
 	})
 }
 
+#[test]
+fn trigger_tasks_updates_executions_left() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		Balances::set_balance(RawOrigin::Root.into(), ALICE, 1000, 5).unwrap();
+		let task_id01 = add_task_to_task_queue(
+			ALICE,
+			vec![40],
+			vec![SCHEDULED_TIME, SCHEDULED_TIME + 3600],
+			Action::NativeTransfer { sender: ALICE, recipient: BOB, amount: 1 },
+		);
+
+		match AutomationTime::get_task(task_id01) {
+			None => {
+				panic!("A task should exist if it was scheduled")
+			},
+			Some(task) => {
+				assert_eq!(task.executions_left(), 2);
+			},
+		}
+
+		LastTimeSlot::<Test>::put((LAST_BLOCK_TIME, LAST_BLOCK_TIME));
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(120_000);
+
+		match AutomationTime::get_task(task_id01) {
+			None => {
+				panic!("A task should exist if it was scheduled")
+			},
+			Some(task) => {
+				assert_eq!(task.executions_left(), 1);
+			},
+		}
+	})
+}
+
+#[test]
+fn trigger_tasks_removes_completed_tasks() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		Balances::set_balance(RawOrigin::Root.into(), ALICE, 1000, 5).unwrap();
+		let task_id01 = add_task_to_task_queue(
+			ALICE,
+			vec![40],
+			vec![SCHEDULED_TIME],
+			Action::NativeTransfer { sender: ALICE, recipient: BOB, amount: 1 },
+		);
+
+		match AutomationTime::get_task(task_id01) {
+			None => {
+				panic!("A task should exist if it was scheduled")
+			},
+			Some(task) => {
+				assert_eq!(task.executions_left(), 1);
+			},
+		}
+
+		LastTimeSlot::<Test>::put((LAST_BLOCK_TIME, LAST_BLOCK_TIME));
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(120_000);
+
+		assert_eq!(AutomationTime::get_task(task_id01), None);
```

Comment:
can you show that the task was run by checking the events as you did above. just makes it easier to read